docker/hq-compose-runserver.yml

```diff
@@ -1,34 +1,53 @@
 version: '2.3'
 
+volumes:
+  shared_files: {}
+
+# x-anchors are ignored by Docker, are used for DRYness here
+x-commcarehq_base: &commcarehq_base
+  image: commcarehq_incl:latest
+  environment: &environment_base
+    COMMCAREHQ_BOOTSTRAP: "yes"
+    DEPENDENT_SERVICES: "couch:5984 postgres:5432 redis:6379 elasticsearch2:9200 minio:9980 kafka:9092"
+    DOCKER_HQ_OVERLAY: "${DOCKER_HQ_OVERLAY}"
+    DOCKER_HQ_OVERLAYFS_CHMOD: "${DOCKER_HQ_OVERLAYFS_CHMOD}"
+    DOCKER_HQ_OVERLAYFS_METACOPY: "${DOCKER_HQ_OVERLAYFS_METACOPY}"
+  privileged: true  # allows mount inside container
+  depends_on:
+    - postgres
+    - couch
+    - redis
+    - elasticsearch2
+    - elasticsearch5
+    - kafka
+    - zookeeper
+    - minio
+  volumes:
+    - shared_files:/sharedfiles
+
 services:
   web:
-    build:
-      context: .
-      dockerfile: "${DOCKERFILE}"
-    environment:
-      COMMCAREHQ_BOOTSTRAP: "yes"
-      DEPENDENT_SERVICES: "couch:5984 postgres:5432 redis:6379 elasticsearch2:9200 minio:9980 kafka:9092"
-      DOCKER_HQ_OVERLAY: "${DOCKER_HQ_OVERLAY}"
-      DOCKER_HQ_OVERLAYFS_CHMOD: "${DOCKER_HQ_OVERLAYFS_CHMOD}"
-      DOCKER_HQ_OVERLAYFS_METACOPY: "${DOCKER_HQ_OVERLAYFS_METACOPY}"
-    privileged: true  # allows mount inside container
-    command: [runserver]
-    depends_on:
-      - postgres
-      - couch
-      - redis
-      - elasticsearch2
-      - elasticsearch5
-      - kafka
-      - zookeeper
-      - minio
+    <<: *commcarehq_base
+    command: 'docker/start-web.sh'
     expose:
       - 8000
     ports:
       - "8000:8000"
-    volumes:
-      - ..:/mnt/commcare-hq-ro:ro
-      - ${VOLUME_PREFIX}lib:/mnt/lib
+
+  celery:
+    <<: *commcarehq_base
+    depends_on:
+      - web
+    command: 'celery -A corehq worker -l info'
+
+  pillowtop:
+    <<: *commcarehq_base
+    environment:
+      <<: *environment_base
+      KAFKA_ADVERTISED_HOST_NAME: kafka
+    depends_on:
+      - web
+    command: './manage.py run_ptop --all --processor-chunk-size=1'
```

Comment:
Is `--all` what you want to do? I think this will spin up _every pillow_ that exists in `settings.py`; which might be every one known to the HQ codebase, including ones that are only designed to run in special scenarios (unless those have been removed since I last tried this). It has been explained to me that this (using `--all`) is not a meaningful pillow configuration for any environment. From personal experience, I think this will at least result in copious amounts of pillow tracebacks.

Personal disclosure: if you find a useful subset of pillows to run on a "bog standard environment", [I'm very interested in learning what that is](https://dimagi.slack.com/archives/C02QG63MU/p1646680380576529?thread_ts=1646332880.768929&cid=C02QG63MU).